app/CurrikiGo/SafariMontage/EasyUpload.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace App\CurrikiGo\SafariMontage;
+
+use App\CurrikiGo\SafariMontage\Client;
+use App\Services\CurrikiGo\LTIConsumerService;
+use App\Models\Activity as ActivityModel;
+use App\Models\CurrikiGo\LmsSetting;
+use App\Models\SsoLogin;
+use Illuminate\Support\Facades\Auth;
+
+/**
+ * EasyUpload class for handling publishing to Safari Montage LMS
+ */
+class EasyUpload
+{
+    /**
+     * LMS Settings modal object
+     * 
+     * @var \App\Models\CurrikiGo\LmsSetting
+     */
+    private $lmsSetting;
+
+    const UPLOAD_API_ENDPOINT = '/SAFARI/api/imslti.php';
+    const PROVIDER_NAME = 'safarimontage';
+    const CONTEXT_TITLE = 'Easy Upload';
+    
+    /**
+     * Make an instance of the class
+     * 
+     * @param \App\Models\CurrikiGo\LmsSetting $lmsSetting
+     */
+    public function __construct(LmsSetting $lmsSetting)
+    {
+        $this->lmsSetting = $lmsSetting;
+    }
+
+    /**
+     * Send a playlist to Safarmi Montage
+     * 
+     * @param \App\Models\Activity $activity
+     * @param array $data 
+     * @return array
+     */
+    public function uploadActivity(ActivityModel $activity, $data)
+    {
+        $user = Auth::user();
+
+        // Prepare LTI request parameters.
+        $key = $this->lmsSetting->lms_access_key;
+        $secret = $this->lmsSetting->lms_access_secret;
+        $launchUrl = $this->lmsSetting->lms_url . '' . self::UPLOAD_API_ENDPOINT;
```

Comment:
Why use (. '' .) for string concatenation ?

Reply:
Fixed.